### PR TITLE
Move OverlappingSubmapTrimmerOptions to PoseGraphOptions

### PR DESCRIPTION
**Pull request by cschuet**

### CODE REVIEW

File: cartographer/mapping/internal/2d/overlapping_submaps_trimmer_2d.h [hunk 1]
```diff
@@ -23,15 +23,20 @@
 namespace cartographer {
 namespace mapping {
 
+/**
```
Comment (gaschler) [inline cartographer/mapping/internal/2d/overlapping_submaps_trimmer_2d.h]: clean up

File: cartographer/mapping/internal/2d/overlapping_submaps_trimmer_2d.h [hunk 2]
```diff
@@ -41,8 +46,8 @@ class OverlappingSubmapsTrimmer2D : public PoseGraphTrimmer {
  private:
   // Number of the most recent submaps to keep.
   const uint16 fresh_submaps_count_;
-  // Minimal number of covered cells to keep submap from trimming.
-  const uint16 min_covered_cells_count_;
+  // Minimum area of covered space to keep submap from trimming.
```
Comment (gaschler) [inline cartographer/mapping/internal/2d/overlapping_submaps_trimmer_2d.h]: mention unit `m^2` in comment

File: cartographer/mapping/internal/2d/overlapping_submaps_trimmer_2d.cc [hunk 3]
```diff
@@ -198,9 +199,9 @@ void OverlappingSubmapsTrimmer2D::Trim(Trimmable* pose_graph) {
                              pose_graph->GetConstraints());
   const std::set<SubmapId> all_submap_ids = AddSubmapsToSubmapCoverageGrid2D(
       submap_freshness, submap_data, &coverage_grid);
-  const std::vector<SubmapId> submap_ids_to_remove =
-      FindSubmapIdsToTrim(coverage_grid, all_submap_ids, fresh_submaps_count_,
-                          min_covered_cells_count_);
+  const std::vector<SubmapId> submap_ids_to_remove = FindSubmapIdsToTrim(
+      coverage_grid, all_submap_ids, fresh_submaps_count_,
+      min_covered_area_ / common::Pow2(coverage_grid.resolution()));
```
Comment (gaschler) [inline cartographer/mapping/internal/2d/overlapping_submaps_trimmer_2d.cc]: nice

**Comment by cschuet** [comment]
@pifon2a FYI

File: cartographer/mapping/internal/2d/overlapping_submaps_trimmer_2d.h (hunk 1 above)
Comment (cschuet) [reply cartographer/mapping/internal/2d/overlapping_submaps_trimmer_2d.h]: done.

File: cartographer/mapping/internal/2d/overlapping_submaps_trimmer_2d.h (hunk 2 above)
Comment (cschuet) [reply cartographer/mapping/internal/2d/overlapping_submaps_trimmer_2d.h]: done.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by cschuet** [comment]
@MichaelGrupp This moves the overlapping submaps trimmer into the PoseGraphOptions. Now you should be able to enable the trimmer just on the server by setting the appropriate options in your server pose_graph lua dictionary.

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Comment by MichaelGrupp** [comment]
Danke!

**Pull request merged**